### PR TITLE
Updated AI scores for Status moves that Dynamax Pokemon are immune to

### PR DESCRIPTION
**Pull request by WillKolada**
The AI will now decrease the scores of moves that don't affect Dynamax Pokemon when they're trying to target Dynamax Pokemon with them.

## **Discord contact info**
Special K#8400

### CODE REVIEW

File: src/battle_ai_util.c [hunk 1]
```diff
@@ -431,6 +431,7 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
             return TRUE;
         break;
     case EFFECT_LOW_KICK:
+    case EFFECT_HEAT_CRASH:
```
Comment (AlexOn1ine) [inline src/battle_ai_util.c:434]: weird that this change is here since it is already in master

Comment (WillKolada) [reply src/battle_ai_util.c:434]: I thought the same thing but didn't question it since it seemed harmless

File: src/battle_ai_main.c [hunk 2]
```diff
@@ -1002,7 +1002,7 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
     // heal block check
     if (gStatuses3[battlerAtk] & STATUS3_HEAL_BLOCK && IsHealBlockPreventingMove(battlerAtk, move))
         return 0; // Can't even select heal blocked move
-
+
```
Comment (Bassoonian) [inline src/battle_ai_main.c:1005]: No point in having these spaces here

File: src/battle_ai_main.c [hunk 3]
```diff
@@ -1434,6 +1434,8 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
                 return 0;
             if (!ShouldTryOHKO(battlerAtk, battlerDef, aiData->abilities[battlerAtk], aiData->abilities[battlerDef], move))
                 ADJUST_SCORE(-10);
+	    else if (IsDynamaxed(battlerDef))
+		ADJUST_SCORE(-10);
```
Comment (Bassoonian) [inline src/battle_ai_main.c:1438]: You're not consistent in adding Dynamax checks as separate conditions (here) or adding them as or conditions to an already existing `if` (see above). I'd choose one and not vary in the same file

**Comment by WillKolada** [comment]
Thanks for the review, Bassoonian. I also want to add that, upon checking `battle_dynamax.c`, none of the status moves are included in `IsMoveBlockedByDynamax`. Should I add those in this PR?

**Comment by AlexOn1ine** [comment]
> `IsMoveBlockedByDynamax`

You could update the list and instead of adding each check individually you could add a check somewhere at the beginning of `AI_CheckBadMove` and return a -10 score.

**Comment by WillKolada** [comment]
Good idea @AlexOn1ine . As for a list, I've just been referencing the "Immunities" section of this Bulbapedia article: https://bulbapedia.bulbagarden.net/wiki/Dynamax#Effects

Cross-referencing it with the following Smogon post seems to corroborate it.
https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/

File: src/battle_ai_util.c (hunk 1 above)
Comment (AsparagusEduardo) [reply src/battle_ai_util.c:434]: It's because this branch hasn't been synced to master after the other PR :)

**Comment by WillKolada** [comment]
> @WillKolada I would just do the AI changes for now. Sorry for the suggestion by me without further looking into the code.

No worries! It makes sense to me as a mini-refactor in the future. Would be a lot clearer to people in the future what moves are negated by Dynamax, rather than digging through battle_script_commands.c to find out. For now, the AI tweaks should be ready to review again.

**Comment by Bassoonian** [comment]
My line 1005 comment still goes. Also, you seem to be using tabs over spaces nearly everywhere, so that needs to be fixed.

**Pull request merged**